smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy

```diff
@@ -556,31 +561,23 @@ def pollResponse(response, data) {
 								currentActivity = getActivityName(it.value.response.data.currentAvActivity,it.key)
 							hub.sendEvent(name: "currentActivity", value: currentActivity, descriptionText: "Current activity is ${currentActivity}", displayed: false)
 						}
+						it.value.response.data.currentActivities.each {currentActivity ->
+							currentActivities.add("harmony-${it.key}-${currentActivity}")
+						}
 					}
 	          	} else {
 	            	log.trace "Harmony - error response: $it.value.message"
 	          	}
         	}
-	        def activities = getChildDevices()
 	        def activitynotrunning = true
+	        log.debug "Harmony - Current Activities: $currentActivities"
 	        activities.each { activity ->
-	            def act = activity.deviceNetworkId.split('-')
-	            if (act.size() > 2) {
-	                def aux = map.find { it.key == act[1] }
-	                if (aux) {
-	                    def aux2 = aux.value.split(',')
-	                    def childDevice = getChildDevice(activity.deviceNetworkId)
-	                    if ((act[2] == aux2[0]) && (aux2[1] == "1" || aux2[1] == "2")) {
-	                        childDevice?.sendEvent(name: "switch", value: "on")
-	                        if (aux2[1] == "1")
-	                            runIn(5, "poll", [overwrite: true])
-	                    } else {
-	                        childDevice?.sendEvent(name: "switch", value: "off")
-	                        if (aux2[1] == "3")
-	                            runIn(5, "poll", [overwrite: true])
-	                    }
-	                }
-	            }
+                def childDevice = getChildDevice(activity.deviceNetworkId)
```

Comment:
Similare comment as above, isn't activities a list of childDevices? so you shouldn't have to do another getChildDevice() on a specific deviceNetworkId here...